app/src/main/java/com/hedvig/app/feature/offer/ui/changestartdate/ChangeDateBottomSheetViewModel.kt

```diff
@@ -57,12 +65,16 @@ class ChangeDateBottomSheetViewModel(
         viewModelScope.launch {
             tracker.changeDateContinue()
             _viewState.value = ViewState.Loading(true)
-            val results = selectedDates.map {
-                editStartDateUseCase.setStartDate(
-                    id = it.key,
-                    idsInBundle = data.idsInBundle,
-                    date = it.value
-                )
+            val results = coroutineScope {
```

Comment:
Why are we opening a new scope here? 🤔

Reply:
This comes from this article (https://elizarov.medium.com/structured-concurrency-722d765aa952) I read from Elizarov the other day.
Jump to the bottom of it or Cmd + F for "You cannot simply write async { … } anymore, you have to provide a scope."
Without it, if one of the async fails, it fails silently, the rest continue, but we do want them all to be done successfully normally right? I can do some testing around this to see how it behaves when something goes bad

Reply:
Fair enough